plugins/modules/system/keyring_info.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, Alexander Hussey <ahussey@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+Ansible Module - community.general.keyring_info
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: keyring_info
+version_added: 5.1.0
+author:
+  - Alexander Hussey (@ahussey-redhat)
+short_description: Get a passphrase using the Operating System's native keyring
+description: >-
+  This module uses the L(keyring Python library, https://pypi.org/project/keyring/)
+  to retrieve passphrases for a given service and username from the OS' native keyring.
+requirements:
+  - keyring (Python library)
+  - gnome-keyring (application - required for headless Linux keyring access)
+  - dbus-run-session (application - required for headless Linux keyring access)
+options:
+  service:
+    description: The name of the service.
+    required: true
+    type: str
+  username:
+    description: The user belonging to the service.
+    required: true
+    type: str
+  keyring_pass:
+    description: Password to unlock keyring.
+    required: true
+    type: str
+"""
+
+EXAMPLES = r"""
+- name: keyring_info module test
+  hosts: localhost
+  connection: local
+  gather_facts: true
+  become: false
+  vars:
+    keyring_pass: test
+  tasks:
+    - name: Retrieve password for service_name/user_name
+      community.general.keyring_info:
+        service: test
+        username: test1
+        keyring_pass: "{{ keyring_pass }}"
+      register: test_password
+      no_log: true
+
+    - name: display password
+      debug:
+        msg: "{{ test_password.result }}"
+"""
+
+RETURN = r"""
+original_message:
+  description: The parameters sent to the module.
+  type: str
+  returned: always
+  sample: 'service_name@user_name'
+result:
+  description: The passphrase for the given service and user.
+  type: str
+  returned: success
+  sample: 'password123'
+"""
+
+try:
+    from shlex import quote
+except ImportError:
+    from pipes import quote
+from subprocess import Popen, PIPE
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+try:
+    import keyring
+
+    HAS_KEYRING = True
+except ImportError:
+    HAS_KEYRING = False
+    KEYRING_IMP_ERR = traceback.format_exc()
+
+
+def _alternate_retrieval_method(module):
+    with Popen(
```

Comment:
This must be removed. Use `module.run_command`. (https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html#module-security)